boot/freeldr/freeldr/arch/i386/multiboot.S

```diff
@@ -91,6 +95,35 @@ MultibootEntry:
     cmp eax, MULTIBOOT_BOOTLOADER_MAGIC
     jne mbfail
 
+    /* Save multiboot info structure */
```

Comment:
Why does all of this need to be in asm?

Reply:
Asm code part is executed initially and then entire freeldr gets relocated. Relocation could rewrite some multiboot structures, that's why it's done here.

Reply:
It probably could be rewritten in C, but it's question to the code maintainer. 🙂

Reply:
I would suggest doing it in C. If it needs to be done here, add a single call to a helper function. That funtion will probably small and much simpler to understand than all of this.

Reply:
I agree, but it's something for later.

Reply:
I agreee, but I'm not sure we can easily jump to C code from here at this intermediate stage, so it's better to convert this to C code later (hope @HBelusca will help).